source/elements/oneDAL/include/oneapi/dal/kmeans_init.hpp

```diff
@@ -0,0 +1,83 @@
+#include "oneapi/dal/table.hpp"
+#include "oneapi/dal/train.hpp"
+
+namespace oneapi::dal::kmeans_init {
+
+namespace method {
+   /// Tag-type that denotes `Default dense`_.
+   struct dense {};
+
+   using by_default = dense;
+} // namespace method
+
+/// @tparam Float  The floating-point type that the algorithm uses for
+///                intermediate computations. Can be :expr:`float` or
+///                :expr:`double`.
+/// @tparam Method – Tag-type that specifies an implementation
+/// of K-Means Initialization algorithm.
+template <typename Float = float,
+          typename Method = method::by_default>
+class descriptor {
+public:
+
+   /// Creates new instance of descriptor with the default attribute values.
+   descriptor();
+
+   /// The number of clusters $k$
+   /// @invariant :expr:`cluster_count > 0`
+   std::int64_t get_cluster_count() const;
+   descriptor& set_cluster_count(std::int64_t);
+
+};
+
+class compute_input {
+public:
+
+   /// Creates input for the computing operation with the default attribute values.
+   /// @remark default = table{}
+   compute_input();
+
+   /// Creates input for the computing operation with the given data,
+   /// the other attributes get default values.
+   compute_input(const table& data = table{});
+
+   /// The computing set $X$
```

Comment:
Not clear to me what this 'computing set' is. The set that the computations are performed on?

Reply:
fixed